### PR TITLE
Enabled loading of additional context sources

### PR DESCRIPTION
**Pull request by LBanovic**

### CODE REVIEW

**Comment by ggozad** [comment]
This sounds like something I would be very interested in. I am not sure how it would fit in `oterm` but unless you are decided you don't want to discuss it, I would be happy to.

**Comment by LBanovic** [comment]
> This sounds like something I would be very interested in. I am not sure how it would fit in `oterm` but unless you are decided you don't want to discuss it, I would be happy to.

So, to preface, I made an accidental pull request and wanted to remove it so maintainers don't get spammed. Instead of a pull request, I attempted to merge that branch with the main of the fork I was working on, but I'm terrible with the Github UI.

With that said, adding context may not be something that `oterm` should handle, but whatever server `oterm` interfaces with. An idea I had was to intercept requests to Ollama and enrich them with context before passing them through. I still think this is the way to go in terms of maintainability and avoiding mixing ML processing directly with `oterm` - data processing doesn't get mixed with data presentation. I believe LangChain has some tools to help with that, but I'm not terribly familiar with their framework so I stuck with the libraries I'm comfortable with.

I opted for an alternate approach, shoe-horning everything into `oterm`, because I had already built a rudimentary prototype of the functionality, but with a terrible UI. So when I saw how superior oterm UI was to my "handrolled" one, I resolved to just integrate some parts of the previous project, and use it like that, for reasons I tried to state in the updated README.md of the fork.

TL;DR: some sort of middleman between oterm and ollama could be better for adding context to requests instead of putting that functionality into one project or the other.

**Comment by ggozad** [comment]
I absolutely agree with what you state here. `oterm` is the wrong product for this. The reason I am interested in exploring further is that I was also thinking of doing a separate project, borrowing perhaps parts of `oterm` and focusing on the RAG parts. I haven't yet anything and my time is very limited, but if you want to start something along these lines please get into contact.
